NuRadioReco/modules/phasedarray/triggerSimulator.py

```diff
@@ -18,265 +19,284 @@
+    det: Detector object
+        Description of the current detector
+    beam_rolls: array of ints
+        Contains the integers for rolling the voltage traces (delays)
+    sec_beam_rolls: array of ints
+        Contains the secondary beam integers for rolling the voltage traces (delays)
+    triggered_channels: array of ints
+        Ids of the triggering channels
+    threshold: float
+        Voltage threshold for a SINGLE antenna. It will be rescaled with the
+        square root of the number of antennas.
+    window_time: float
+        Width of the time window used in the power integration
+    cut_times: (float, float) tuple
+        Times for cutting the trace. This helps reducing the number of noise-induced triggers.
+    trigger_adc: bool
+        If True, analog to digital conversion is performed. It must be specified in the
+        detector file. See analogToDigitalConverter module for information
+    upsampling_factor: integer
+        Upsampling factor. The trace will be a upsampled to a
+        sampling frequency int_factor times higher than the original one
+        before conversion to digital
+    nyquist_zone: integer
+        To be used with the trigger_adc function
+        If None, the trace is not filtered
+        If n, it uses the n-th Nyquist zone by applying an 8th-order
+        Butterworth filter with critical frequencies:
+        (n-1) * adc_sampling_frequency/2 + bandwidth_edge
+        and
+        n * adc_sampling_frequency/2 - bandwidth_edge
+    bandwidth_edge: float
+        Frequency interval used for filtering the chosen Nyquist zone.
+        See above
+
+    Returns
+    -------
+    is_triggered: bool
+        True if the triggering condition is met
+    trigger_delays: dictionary
+        the delays for the primary channels that have caused a trigger.
+        If there is no trigger, it's an empty dictionary
+    sec_trigger_delays: dictionary
+        the delays for the secondary channels that have caused a trigger.
+        If there is no trigger or no secondary channels, it's an empty dictionary
+    """
+    station_id = station.get_id()
+    beam_rolls = get_beam_rolls(station, det, triggered_channels, phasing_angles,
+                                ref_index=ref_index, trigger_adc=trigger_adc,
+                                upsampling_factor=upsampling_factor)
+
+    secondary_beam_rolls = [{} for direction in range(len(phasing_angles))]
+    if(len(secondary_channels) > 0):
+        secondary_beam_rolls = get_beam_rolls(station, det, secondary_channels, secondary_phasing_angles,
+                                              ref_index=ref_index, trigger_adc=trigger_adc,
+                                              upsampling_factor=upsampling_factor)
+
+    traces = {}
+    for channel_id in triggered_channels:
+        channel = station.get_channel(channel_id)
+        channel_id = channel.get_id()
+        time_step = 1 / channel.get_sampling_rate()
+
+        if trigger_adc:
```

Comment:
Would be nice to explain what is happening here. I think what what you're doing is getting all the traces pulled into a general array? But it would be good to just explain that so the reader isn't guessing.